homeassistant/components/rainmachine/switch.py

```diff
@@ -164,6 +198,28 @@ async def async_setup_entry(
             },
             "async_start_zone",
         ),
+        (
+            "push_weather_data",
+            {
+                vol.Optional(ATTR_TIMESTAMP, default=None): vol.Any(
```

Comment:
Do we need `vol.Any(cv.positive_float, None)`? I would think that since we mark this field as optional, `None` is already a valid value. Wondering if this would work:

```python
vol.Optional(ATTR_TIMESTAMP, default=None): cv.positive_float
```